account_invoice_date_due/views/account_move_date_due.xml

```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record id="view_move_form" model="ir.ui.view">
+        <field name="name">account.move.form</field>
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="account.view_move_form" />
+        <field name="arch" type="xml">
+            <field name="invoice_date_due" position="before">
+                <field name="is_invoice_date_due_editable" invisible="1" />
+            </field>
+            <field name="invoice_date_due" position="attributes">
+                <attribute name="readonly">0</attribute>
```

Comment:
Wouldn't it be easier to play with fields declared multiple times and positive/negative groups? Something like this for the regular field:
```xml
<attribute name="groups">!account_invoice_date_due.allow_to_change_due_date</attribute>
```
And then:
```xml
<field name="invoice_date_due" position="after">
    <field name="invoice_date_due" groups="account_invoice_date_due.allow_to_change_due_date" attrs="{'readonly': your domain, 'invisible': another domain}" />
</field>
```